src/components/Commits/CommitsListView.tsx

```diff
@@ -77,6 +77,12 @@ const CommitsListView: React.FC<CommitsListViewProps> = ({
     ),
   );
 
+  React.useEffect(() => {
+    if (loaded && !plrError && getNextPage) {
+      getNextPage();
+    }
+  }, [loaded, plrError, getNextPage]);
```

Comment:
This simply over fetches all the pipelineruns in the workspace even if the user isn't scrolling down the commit list view. I think its better to fire this `getNextPage` only when the User hits the bottom of the table just like we do for pipelinerunListView.

Reply:
The fetching on page scroll would work only if the commits were paged. If we don't get all the pipeline runs, we may not find the run associated with a commit that is shown.